src/blog/2024/06/filtering-mapping-sorting-reducing-with-node-red.md

```diff
@@ -0,0 +1,167 @@
+---
+title: How to Filter, Map, Sort, and Reduce Data in Node-RED
+subtitle: Step-by-step guide on how to perform filtering, mapping, sorting and reducing in Node-RED.
+description: Learn how to perform data transformation in Node-RED with a low-code approach.
+date: 2024-06-19
+authors: ["sumit-shinde"]
+image: 
+tags:
+   - posts
+   - node-red
+   - flowfuse
+   - low-code
+   - filtering
+   - sorting
+   - mapping 
+   - reducing
+   - data tranformation
+---
+   
+[Node-RED](/node-red/) is a visual programming platform known for its user-friendly interface and low-code approach, that allows developers to spend more time developing solutions instead of dealing with technical complexities. However, users can sometimes make applications developed with Node-RED complex and hard to maintain when they are not familiar with its basics. Understanding Node-RED basics helps keep flows low-coded.
+
+<!--more-->
+
+Often, I have noticed that many users are unfamiliar with how to filter, map, sort, and reduce data using a low-code approach. They tend to rely on function nodes, which can add complexity. If you are unfamiliar with function nodes and their drawbacks, please refer to [The benefits and drawbacks of using Node-RED function nodes](/blog/2023/03/why-should-you-use-node-red-function-nodes/). In this guide, we will show you how to filter, map, sort, and reduce data with Node-RED while sticking to low-code practices.
+
+## What is Low-Code
+
+Low-code is a software development approach that requires little to no coding to build applications and processes. Instead of using complex programming languages, you use visual interfaces with basic logic and drag-and-drop capabilities.
+
+> Low-code is not just about accelerating development; it’s about democratizing it. It’s about giving more people the ability to create solutions to business problems.
+>
+> *— Charles Lamanna, Corporate Vice President of Business Applications & Platforms at Microsoft*
+
+For more details refer to the following guides:
+
+- [Why Low-Code is Better](/blog/2024/03/low-code-is-better/).
+- [Why you need a low-code platform](/blog/2024/05/why-you-need-a-low-code-platform/).
+
+
+## Why do you need to learn to filter, map, sort, and reduce the data?
+
+Learning to filter, map, sort, and reduce data is crucial for effective data processing and transformation. For example, consider the scenario where you have an array of sensor data retrieved from an InfluxDB database. The data looks something like this:
+
+```json
+[
```

Comment:
Yes, you could map from `emperature` to `temperature`, but maybe better to map from Kelvin to Celsius, or so?

Reply:
@sumitshinde-84 Please update this